File: examples/availability/publish_hosted/PublishHosted.java
import com.bloock.sdk.Bloock;
import com.bloock.sdk.client.AvailabilityClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.availability.HostedPublisher;
import com.bloock.sdk.entity.record.Record;

public class PublishHosted {
  public static void main(String[] args) throws Exception {
    try {
      Bloock.apiKey = System.getenv("API_KEY");

      RecordClient recordClient = new RecordClient();
      AvailabilityClient availabilityClient = new AvailabilityClient();

      Record record = recordClient.fromString("Hello world").build();

      String id = availabilityClient.publish(record, new HostedPublisher()); // it returns the hash
    } catch (Exception e) {
      System.out.println(e);
    }
  }
}

File: examples/availability/publish_hosted/publish_hosted.go
package main

import (
	"fmt"
	"log"
	"os"

	bloock "github.com/bloock/bloock-sdk-go/v2"
	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity/availability"
)

func main() {
	bloock.ApiKey = os.Getenv("API_KEY")

	recordClient := client.NewRecordClient()
	availabilityClient := client.NewAvailabilityClient()

	record, err := recordClient.FromString("Hello world").Build()
	if err != nil {
		log.Fatalln(err)
	}

	_, err = availabilityClient.Publish(record, availability.NewHostedPublisher())
	if err != nil {
		log.Fatalln(err)
	}

	fmt.Println("Record was published successfully")
}
